### PR TITLE
Add document on testing RISC-V in qemu

### PR DESCRIPTION
**Pull request by laanwj**
(don't know if this is the right place, I usually use gists for things like this, this is an experiment in actually using the `docs` repo)

### CODE REVIEW

**Comment by MarcoFalke** [comment]
An alternative would be to use the https://github.com/bitcoin-core/bitcoin-devwiki/wiki

**Comment by laanwj** [comment]
The wiki can be editing by everyone; I considered that, but found that too scary as long as it encourages downloading something.

**Comment by MarcoFalke** [comment]
Good point.

Approach ACK

**Comment by fanquake** [comment]
Concept ACK, thanks for writing this up. I will read through shortly, and actually install / test in a day or two.

File: testing/qemu_riscv.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+Fedora rawhide) here:
+
+```bash
+mkdir -p /usr/gnemul
+cd /usr/gnemul
+wget https://dev.visucore.com/bitcoin/qemu-riscv64-libs.tar.gz
+tar -zxvf qemu-riscv64-libs.tar.gz
+```
+
+(TODO: get this files directly from the packages instead of from wumpus' server)
```
Comment (laanwj) [inline testing/qemu_riscv.md]: Some distributions might even already have a package that provides these, for cross-architecture. e.g. https://wiki.debian.org/RISC-V mentions:
```
$ sudo dpkg --add-architecture riscv64
$ sudo apt-get install gcc-riscv64-linux-gnu g++-riscv64-linux-gnu
$ sudo sh -c "cat >/etc/ld.so.conf.d/riscv64-linux-gnu.conf <<EOF
/usr/local/lib/riscv64-linux-gnu
/lib/riscv64-linux-gnu
/usr/lib/riscv64-linux-gnu
/usr/riscv64-linux-gnu/lib/
EOF
"
$ sudo ln -s /usr/riscv64-linux-gnu/lib/ld-linux-riscv64-lp64d.so.1 /lib
```
I haven't tried this.

File: testing/qemu_riscv.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+To make them available you could download the archive of previous section and do:
+
+```bash
+ln -s /usr/gnemul/qemu-riscv64/lib64 /lib64/lp64d
+```
+
+Launching bitcoin
+------------------
+
+After this, bitcoin, or the unit tests, can be launched as normal with:
```
Comment (fanquake) [inline testing/qemu_riscv.md]: nit: s/bitcoin/bitcoind

File: testing/qemu_riscv.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+These libraries are necessary:
+
+    libm.so.6 -> libm-2.29.9000.so
+    libpthread.so.0 -> libpthread-2.29.9000.so
+    libatomic.so.1 -> libatomic.so.1.2.0
+    libgcc_s.so.1 -> libgcc_s-9-20190503.so.1
+    libc.so.6 -> libc-2.29.9000.so
+
+To make them available you could download the archive of previous section and do:
```
Comment (fanquake) [inline testing/qemu_riscv.md]: nit: s/of/from the

**Comment by MarcoFalke** [comment]
Closing due to inactivity

**Comment by laanwj** [comment]
I'm fine with closing inactive PRs (though in this case a reminder would have been helpful too, I just forgot about this). But why do you delete my branch too?

**Comment by Sjors** [comment]
Yeah, please don't delete branches. I recently almost lost a WIP branch that I didn't have on a new computer (though fortunately it was still on Github).

**Comment by MarcoFalke** [comment]
Sorry for closing without warning, reopened for now. (Branches that were in a pull request can be restored on GitHub)

File: testing/qemu_riscv.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+These same steps work for other architectures but for sake of simplicity, I'll
+only focus on RISC-V here. This comes up a lot for RISC-V specifically because
+it's still so hard to find hardware for it.
+
+Installing qemu
+----------------
+
+Many distributions don't provide packages for a qemu that supports RISC-V 
+as upstream support for this has been merged somewhat recently. Here are some steps
+to build one from source.
```
Comment (MarcoFalke) [inline testing/qemu_riscv.md]: It is available on `cosmic` and newer, right?

**Comment by laanwj** [comment]
I don't have motivation to work on this anymore, so closing. Anyone interested in picking this up, feel free.

**Pull request closed without merging**